iteration/iteration_test.go

```diff
@@ -38,289 +38,273 @@ func (s *TestIterationRepository) TearDownTest() {
-
-	i2L, err := repo.Load(context.Background(), i2.ID)
-	require.Nil(t, err)
-	assert.NotEmpty(t, i2.Path)
-	i2.Path.Convert()
-	expectedPath := i2.Path.Convert()
-	require.NotNil(t, i2L)
-	assert.Equal(t, expectedPath, i2L.Path.Convert())
-}
+		i := iteration.Iteration{
+			Name:    name,
+			SpaceID: fxt.Spaces[0].ID,
+			StartAt: &start,
+			EndAt:   &end,
+		}
 
-func (s *TestIterationRepository) TestRootIteration() {
-	t := s.T()
-	resource.Require(t, resource.Database)
+		repo.Create(context.Background(), &i)
+		if i.ID == uuid.Nil {
+			t.Errorf("Iteration was not created, ID nil")
+		}
 
-	repo := iteration.NewIterationRepository(s.DB)
+		if i.CreatedAt.After(time.Now()) {
+			t.Errorf("Iteration was not created, CreatedAt after Now()?")
+		}
+		assert.Equal(t, start, *i.StartAt)
+		assert.Equal(t, end, *i.EndAt)
+		assert.Equal(t, name, i.Name)
+	})
+
+	t.Run("success - create child", func(t *testing.T) {
+		start := time.Now()
+		end := start.Add(time.Hour * (24 * 8 * 3))
+		name := "Sprint #24"
+		name2 := "Sprint #24.1"
+
+		fxt := tf.NewTestFixture(s.T(), s.DB, tf.Spaces(1))
 
-	start := time.Now()
-	end := start.Add(time.Hour * (24 * 8 * 3))
-	name := "Sprint #24"
-	name2 := "Sprint #24.1"
-
-	fxt := tf.NewTestFixture(s.T(), s.DB, tf.Spaces(1))
-
-	i := iteration.Iteration{
-		Name:    name,
-		SpaceID: fxt.Spaces[0].ID,
-		StartAt: &start,
-		EndAt:   &end,
-	}
-	repo.Create(context.Background(), &i)
-
-	parentPath := append(i.Path, i.ID)
-	require.NotNil(t, parentPath)
-	i2 := iteration.Iteration{
-		Name:    name2,
-		SpaceID: fxt.Spaces[0].ID,
-		StartAt: &start,
-		EndAt:   &end,
-		Path:    parentPath,
-	}
-	repo.Create(context.Background(), &i2)
-
-	res, err := repo.Root(context.Background(), fxt.Spaces[0].ID)
-	require.Nil(t, err)
-	assert.Equal(t, i.Name, res.Name)
-	assert.Equal(t, i.ID, res.ID)
-	expectedPath := i.Path.Convert()
-	require.NotNil(t, res)
-	assert.Equal(t, expectedPath, res.Path.Convert())
+		i := iteration.Iteration{
+			Name:    name,
+			SpaceID: fxt.Spaces[0].ID,
+			StartAt: &start,
+			EndAt:   &end,
+		}
+		repo.Create(context.Background(), &i)
+
+		parentPath := append(i.Path, i.ID)
+		require.NotNil(t, parentPath)
+		i2 := iteration.Iteration{
+			Name:    name2,
+			SpaceID: fxt.Spaces[0].ID,
+			StartAt: &start,
+			EndAt:   &end,
+			Path:    parentPath,
+		}
+		repo.Create(context.Background(), &i2)
+
+		i2L, err := repo.Load(context.Background(), i2.ID)
+		require.Nil(t, err)
+		assert.NotEmpty(t, i2.Path)
+		i2.Path.Convert()
+		expectedPath := i2.Path.Convert()
+		require.NotNil(t, i2L)
+		assert.Equal(t, expectedPath, i2L.Path.Convert())
+	})
+
+	t.Run("same iteration name within space", func(t *testing.T) {
+		name := "Iteration name test"
+
+		fxt := tf.NewTestFixture(s.T(), s.DB,
+			tf.Spaces(2),
+			tf.Iterations(1, tf.SetIterationNames(name)),
+		)
+
+		i := *fxt.Iterations[0]
+		space2 := *fxt.Spaces[1]
+
+		// another iteration with same name within same sapce, should fail
```

Comment:
could you split those 2 tests (pass/fail) with 2 subtests ?

Reply:
yes done in  https://github.com/fabric8-services/fabric8-wit/pull/1719/commits/2949ad3902c40f46bca32e444c41e2c67eb4f458